Src/Extern/HYPRE/AMReX_HypreABecLap3.cpp

```diff
@@ -0,0 +1,503 @@
+                     int max_iter_, const BndryData& _bndry) {
+  bd = _bndry;
+
+  loadMatrix();
+  finalizeMatrix();
+  loadVectors(soln, rhs);
+  finalizeVectors();
+  setupSolver(rel_tol_, abs_tol_, max_iter_);
+  solveDoIt();
+  getSolution(soln);
+  clearSolver();
+}
+
+void HypreABecLap3::loadBndryData(LinOpBCType bc_type, Real bc_value) {
+  const int comp = 0;
+  const Real* dx = geom.CellSize();
+  for (int n=0; n< BL_SPACEDIM; ++n) {
+    for (MFIter mfi(acoefs); mfi.isValid(); ++mfi) {
+      int i = mfi.index();
+
+      const Box& bx = mfi.validbox();
+
+      // Our default will be that the face of this grid is
+      // either touching another grid
+      // across an interior boundary or a periodic boundary.
+      // We will test for the other
+      // cases below.
+      {
+        // Define the type of boundary conditions to be Dirichlet
+        // (even for periodic)
+        bd.setBoundCond(Orientation(n, Orientation::low), i,
+                        comp, LO_DIRICHLET);
+        bd.setBoundCond(Orientation(n, Orientation::high), i,
+                        comp, LO_DIRICHLET);
+
+        // Set the boundary conditions to the
+        // cell centers outside the domain
+        bd.setBoundLoc(Orientation(n, Orientation::low), i, 0.5*dx[n]);
+        bd.setBoundLoc(Orientation(n, Orientation::high), i, 0.5*dx[n]);
+      }
+
+      // Now test to see if we should override
+      // the above with Dirichlet or Neumann physical bc's
+      if(bc_type != LinOpBCType::interior) {
+        int ibnd = static_cast<int>(bc_type);
+        // either LO_DIRICHLET or LO_NEUMANN
+
+        // We are on the low side of the
+        // domain in coordinate direction n
+
+        if (bx.smallEnd(n) == geom.Domain().smallEnd(n)) {
+          // Set the boundary conditions to
+          // live exactly on the faces of the domain
+          bd.setBoundLoc(Orientation(n, Orientation::low), i, 0.0);
+
+          // Set the Dirichlet/Neumann boundary values
+          bd.setValue(Orientation(n, Orientation::low), i, bc_value);
+
+          // Define the type of boundary conditions
+          bd.setBoundCond(Orientation(n, Orientation::low),
+                          i, comp, ibnd);
+        }
+
+        // We are on the high side of the
+        // domain in coordinate direction n
+        if (bx.bigEnd(n) == geom.Domain().bigEnd(n)) {
+          // Set the boundary conditions to
+          // live exactly on the faces of the domain
+          bd.setBoundLoc(Orientation(n, Orientation::high), i, 0.0);
+
+          // Set the Dirichlet/Neumann boundary values
+          bd.setValue(Orientation(n, Orientation::high), i, bc_value);
+
+          // Define the type of boundary conditions
+          bd.setBoundCond(Orientation(n, Orientation::high),
+                          i, comp, ibnd);
+        }
+      }
+    }
+  }
+}
+
+void HypreABecLap3::loadMatrix() {
+  static_assert(BL_SPACEDIM > 1, "HypreABecLap2: 1D not supported");
```

Comment:
HypreABecLap2 --> 3

Reply:
All very good suggestions. Made the changes and will push the code now.